module/state_synchronization/indexer/indexer.go

```diff
@@ -0,0 +1,98 @@
+package indexer
+
+import (
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data/cache"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/jobqueue"
+	"github.com/onflow/flow-go/module/state_synchronization/requester/jobs"
+	"github.com/onflow/flow-go/storage"
+)
+
+const (
+	workersCount = 1 // how many workers will concurrently process the tasks in the jobqueue
+	searchAhead  = 1 // how many block heights ahead of the current will be requested and tasked for jobqueue
+)
+
+// Indexer handles ingestion of new execution data available and uses the execution data indexer module
+// to index the data.
+// The processing of new available data is done by creating a jobqueue that uses the execution data reader to
+// obtain new jobs. The worker also implements the `highestConsecutiveHeight` method which is used by the execution
+// data reader, so it doesn't surpass the highest sealed block height when fetching the data.
+// The execution state worker has a callback that is used by the upstream queues which download new execution data to
+// notify new data is available and kick off indexing.
+type Indexer struct {
+	component       *jobqueue.ComponentConsumer
```

Comment:
I had it emebeeded before and someone suggested not exposing all the methods of the component so I switched. But changing to component type instead of jobqueue type that I can change. But also can embeed again just saying it was before like that.

Reply:
yea, my comment before was about not needing to expose a bunch of methods from the `jobqueue.ComponentConsumer` object. Sorry for not being more clear.